forem/forem
# Replace Pusher Beams with Rpush PN delivery ✂️ ✂️

## What type of PR is this? (check all applicable)


## Description

This PR:
- Removes the `mobile_notifications` FeatureFlag
- Removes Pusher Beams dependency + related logic
- Adds a page to the docs about Rpush explaining the role `ConsumerApp` and `Device` models play in delivering Push Notifications to (mobile) apps.

## Related Tickets & Documents

https://github.com/forem/rfcs/pull/68

## QA Instructions, Screenshots, Recordings

1. Boot app locally and check out `/admin/apps/consumer_apps`
2. CRUD operations should work on `ConsumerApp` records
3. The official Forem iOS app should appear visible but **not operational**
4. If you add `RPUSH_IOS_PEM=anything` to your `.env` file and reboot your local site then **it should appear as operational** (because it detects the credentials from the ENV variable)

Screenshot of PN delivery working in Forem Cloud instances (not just [BHC](https://community.benhalpern.com))

![Image from iOS](https://user-images.githubusercontent.com/6045239/117010511-1b74f680-acaa-11eb-9da7-b87c30031adb.png)

### UI accessibility concerns?

N/A

## Added tests?

- [ ] No, and this is why: We already have test coverage and this PR removes FeatureFlag specific & legacy (Pusher Beams) tests

## [Forem core team only] How will this change be communicated?

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      or in a [forem.dev](http://forem.dev) post

## [optional] Are there any post deployment tasks we need to perform?

None other than make sure Devices start to appear as registered and PNs are being delivered

## [optional] What gif best describes this PR or how it makes you feel?

![rocket cowboy](https://media.giphy.com/media/HtaGVNHVnTNuw/giphy.gif)
